pkg/store/storepb/rpc.proto

```diff
@@ -114,6 +115,11 @@ message SeriesRequest {
 
   // skip_chunks controls whether sending chunks or not in series responses.
   bool skip_chunks = 8;
+
+  // hints_enabled controls whether sending hints or not in the series responses. Hints
```

Comment:
Do we really need this? I would avoid extending API if possible. Especially bool options.

The alternative (which actually avoid lots of boilerplate) is to just add hidden flag for store GW to enable sending hints in the message. This will satisfy all needs and will be perfectly compatible without Store API noise

Reply:
I've updated it based on our offline discussion:
- Removed `hints_enabled` field from `SeriesRequest`
- We'll have an internal flag in `NewBucketStore()` in a subsequent PR (the one that will add hints support to bucket store)